src/backend/vulkan/src/lib.rs

```diff
@@ -895,7 +910,23 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
                 || self.properties.device_id & info::intel::DEVICE_SKY_LAKE_MASK
                     == info::intel::DEVICE_SKY_LAKE_MASK);
 
-        let features = unsafe { self.instance.0.get_physical_device_features(self.handle) };
+        let mut descriptor_indexing_features = None;
+        let features = if let Some(ref get_device_properties) = self.instance.2 {
+            let features = vk::PhysicalDeviceFeatures::builder().build();
+            let mut features2 = vk::PhysicalDeviceFeatures2KHR::builder().features(features).build();
+            if self.supports_extension(*EXT_DESCRIPTOR_INDEXING) {
+                descriptor_indexing_features = Some(vk::PhysicalDeviceDescriptorIndexingFeaturesEXT::builder().build());
+                if let Some(ref mut mut_ref) = descriptor_indexing_features {
+                    mut_ref.p_next = mem::replace(&mut features2.p_next, mut_ref as *mut _ as *mut _);
+                }
+            }
+            unsafe { get_device_properties.get_physical_device_features2_khr(self.handle, &mut features2 as *mut _); }
```

Comment:
can we write it down in a comment how the sequence looks like?
I see `PhysicalDeviceFeatures2KHR` -> `PhysicalDeviceDescriptorIndexingFeaturesEXT` -> `PhysicalDeviceFeatures`. Is that correct?
The code could have been a little simpler if it didn't use `mem::replace`. SInce you know what the current tail is, you should be able to just do `p_next = ...` there.